_posts/2025-04-09-#31664.md

```diff
@@ -10,17 +10,120 @@ status: upcoming
+
+The architecture establishes a clear hierarchy:
+
+1. **`FeeRateForecasterManager`** sits at the top level, coordinating all fee rate forecasters.
+2. **Both `CBlockPolicyEstimator` and `MempoolForecaster`** implement the `Forecaster` interface, providing different approaches to fee rate forecasts.
+
+```
+                           ┌─────────────────┐
+                           │FeeRateForecaster│
+                           │    Manager      │
+                           └───────┬─────────┘
+                                   │ holds a shared_ptr copy of
+                     ┌─────────────┴───────────────┐
+                     │                             │
+            ┌────────▼─────────┐         ┌─────────▼────────┐
+            │CBlockPolicy      │         │Mempool           │
+            │Estimator         │         │Forecaster        │
+            └──────────────────┘         └──────────────────┘
+            (Uses historical mempool data)    (Uses current mempool data)
+```
+
+3. **`ForecastResult` and `ConfirmationTarget`** standardize the input and output formats across all forecasters.
+4. **The node context** maintains the lifecycle of these components, with `ValidationInterface` ensuring they receive blockchain updates.
+
+
+### Design Goals of `#31664`
+
+1. Creates a pluggable architecture for multiple fee estimation strategies.
+2. Creates the pathway for making fee estimation in Bitcoin Core both mempool-aware and package-aware.
+3. Maintains backward compatibility with existing code.
+
+Previous work and discussions to dig deep:
+
+- [PR #23074 attempts to make the fee estimator package aware](https://github.com/bitcoin/bitcoin/pull/23074)
+- [PR #25380 builds on #23074 to just ignore CPFP'd transactions](https://github.com/bitcoin/bitcoin/pull/25380)
+- [PR #30079 is yet another attempt at #25380](https://github.com/bitcoin/bitcoin/pull/30079)
+- [Delving Bitcoin Post: Package-Aware Fee Estimator](https://delvingbitcoin.org/t/package-aware-fee-estimator-post-cluster-mempool/312)
+- [Issue #27995 by Sipa on improving fee estimation](https://github.com/bitcoin/bitcoin/issues/27995)
+- [Blog Post Discussing challenges of fee estimation](https://hackmd.io/@kEyqkad6QderjWKtcBF5Hg/cChallengies-with-estimating-transaction-fees)
+- [Delving Bitcoin Post: Mempool-Based Fee Estimation](https://delvingbitcoin.org/t/mempool-based-fee-estimation-on-bitcoin-core/703)
+- [Empirical Data Analysis of Fee Rate Forecasters for ASAP (next-block) Fee Estimation](https://delvingbitcoin.org/t/empirical-data-analysis-of-fee-rate-forecasters-for-asap-next-block-fee-estimation/1022)
+
+## Questions
+### **Conceptual & Approach**
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
+2. Why is the new system called a "Forecaster" and "ForecasterManager" rather than an "Estimator" and "Fee Estimation Manager"?
```

Comment:
Perhaps add a question or two about the current situation so that people understand the problems.
It could also be helpful to briefly go over the current fee estimation design.

Reply:
I added more notes on the current estimator, Did not think of any more question.
Will push when I do, also happy to take explicit suggestions.